spec/unit/transaction/report_spec.rb

```diff
@@ -136,6 +136,36 @@
     expect(report.metrics['time'].values.any? {|metric| metric.first =~ /whit/i}).to be_falsey
   end
 
+  describe "when exclude_unchanged_resources is true" do
+    before do
+      @test_dir = Dir.mktmpdir
+    end
```

Comment:
There's helper for creating and deleting tmp files/directories:

```suggestion
let(:test_dir) { mktmpdir('unchanged_resources') }
```

It'll automagically work because earlier in this file we do:

https://github.com/puppetlabs/puppet/pull/8947/files#diff-6ece161f30e901a032cac5338d78b06dfe14c12df8050496998faf2293c907faR9